### ISSUE TITLE
Change license to an open source license

### ISSUE THREAD

**Issue by volt4ire** [objective-see/LuLu#4]
The non-commercial clause you use in the license file means the project is not truly "open source" ([see point 6 here](https://opensource.org/docs/definition.php)). Would you consider relicensing it under a truly open source license, like GPL, MIT or CC0? Thanks!

**Comment by JayBrown** [comment]
"NonCommercial" as per CC BY-NC 4.0 is not the same as "No Discrimination Against Fields of Endeavor" in the OSD.

NonCommercial: 

> NonCommercial means not primarily intended for or directed towards commercial advantage or monetary compensation. For purposes of this Public License, the exchange of the Licensed Material for other material subject to Copyright and Similar Rights by digital file-sharing or similar means is NonCommercial provided there is no payment of monetary compensation in connection with the exchange.

Non-Discriminatory:

> The license must not restrict anyone from making use of the program in a specific field of endeavor. For example, it may not restrict the program from being used in a business, or from being used for genetic research.

So a sys admin of a large corporation can install LuLu on all Mac computers in his company, for the improvement of corporate security, i.e. in a business context, but he is not allowed to take LuLu and sell it for his own financial gain.

(My 2¢)

**Comment by volt4ire** [comment]
The OSD specifically states 

> "The major intention of this clause is to **prohibit** license traps that prevent open source from being used **commercially**"

 (emphasis added)

**Comment by JayBrown** [comment]
As for the OSD annotation itself, the *intention* behind a license clause is not relevant. Only the *content* of that clause is. As for the actual content: all they're saying is "use in a business", they don't say "exploit" or "sell" or "distribute" or "import" or "export" etc. *Using* a software is different from *selling* a software. They even offer a specific example, namely "use for genetic research", which seems to suggest that clause 6 does not talk about taking open source software and selling it, but only talks about *applying* ("using") a software in commercial, scientific and other contexts. The CC BY-NC 4.0 doesn't prohibit that; it specifically prohibits "commercial advantage" and "monetary compensation" in the context of "exchanging" (i.e. selling) the product, but that's not what OSD clause 6 seems to be about.

**Comment by JayBrown** [comment]
CC BY-NC 4.0 may however breach OSD clause 1:

> 1. Free Redistribution
> The license shall not restrict any party from selling or giving away the software as a component of an aggregate software distribution containing programs from several different sources. The license shall not require a royalty or other fee for such sale.

But I'm not sure: maybe NonCommercial in CC BY-NC 4.0 only pertains to selling LuLu itself. Then this is not a problem.

**Comment by nicsuzor** [comment]
If the concern is that someone may take the code and sell it, many people use one of the copyleft licences (e.g. GPL) to avoid this problem. It doesn't stop people selling a packaged version, but it requires them to give their code improvements back and keep the source open, so everyone wins.

**Comment by cmsj** [comment]
FWIW, the CC licenses are intentionally vague about what NC actually means, but it's perhaps worth noting that CC discourages the use of their licenses for software: https://creativecommons.org/faq/#can-i-apply-a-creative-commons-license-to-software

I'm also sure that Patrick knows what he's doing, and regardless, this is his copyrighted work that he can license however he sees fit :)

**Comment by jle64** [comment]
OSI's FAQ makes it pretty clear what their view on the subject is :

> Can Open Source software be used for commercial purposes?
>
>   Absolutely. All Open Source software can be used for commercial purpose; the Open Source Definition guarantees this. **You can even sell Open Source software.**

If you don't want to change the license, please consider using a more appropriate term to describe the software, such as "source available".

**Comment by pygy** [comment]
It's not free/libre as described by the FSF, for sure, but I can totally see where @objective-see's coming from with the license too.

The source is available, it can be modified and re-distributed, but not sold... That's still a lot of freedom, and the OSI doesn't own "Open Source".

"Non-commercial, open source firewall" would seem fit.

**Comment by TheLastProject** [comment]
I just noticed this project, and I am sorry, but calling this Open Source with a non-commercial license is simply wrong.

There's a reason it's not listed on the [OSI license list](https://opensource.org/licenses/alphabetical) nor the [FSF license list](https://www.gnu.org/licenses/license-list.html). In fact, the FSF specifically lists it as non-Free (aka: not Open Source).

I also unfortunately have to say @JayBrown is simply wrong on the distinction between Open Source and Free Software. If you read both definitions, you will see that they mean the same in practice. This is logical, because "Open Source" is simply a rebrand of "Free Software" because, according to the OSI, "Free Software" sounds "too political". In practice, there is no difference between the two.

I'd also like to urge to, at very least, not call the project Open Source when it isn't. If you want to use a similar term without changing the license, I'd very much like to request either "Source viewable" or "Shared Source" (which is a Microsoft term to describe, among other things, closed source software where you can view the source code: https://en.wikipedia.org/wiki/Shared_source)

**Comment by objective-see** [comment]
So I learned a lot thru these discussions - mahalo to all, for your thoughtful and insightful comments! My main motivation for making the source of LuLu available was to facilitate transparency. With source, I figured it would allow anybody to take a look at it to see what's going on, or easily audit it. And also, for ppl to contribute code!  At the moment I've decided to keep the license as is, with the non-commercial clause -  but will no longer refer to the project as 'open source'
